cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/AbstractTransactionContext.java

```diff
@@ -152,9 +152,27 @@ public void abort(@Nullable TransactionFailureException cause) throws Transactio
         }
       }
       if (success) {
-        txClient.abort(currentTx);
+        try {
+          txClient.abort(currentTx);
+        } catch (Throwable t) {
+          if (cause == null) {
+            cause = new TransactionFailureException(String.format(
+              "Error while calling transaction service to abort transaction %d.", currentTx.getTransactionId()));
+          } else {
+            cause.addSuppressed(t);
+          }
+        }
       } else {
-        txClient.invalidate(currentTx.getTransactionId());
+        try {
+          txClient.invalidate(currentTx.getTransactionId());
+        } catch (Throwable t) {
```

Comment:
You can move the try-catch to cover the whole of-else to reduce duplicated code. In the catch you can based on the the value of success to alter the msg